### PR TITLE
Model testing

### PR DESCRIPTION
**Pull request by lilwillifo**
# Changes Proposed:
- Ship Placer Spec
-
-
# Fixes:
-
-

# Checklist:
* Did you add to application.yml? no
* Passing RSPEC?  4 failures
* Passing Spec Harness? up to line 46
* Ran on Local Host?

# Mentions:

### CODE REVIEW

File: spec/requests/ship_placer_spec.rb [hunk 1]
```diff
@@ -1,32 +1,42 @@
...
+    end
+    it 'A user cant place a ship diagonally' do
+      headers = {"X-API-Key" => game.player_1.api_key}
+      ship_2_payload = {
+        ship_size: 3,
+        start_space: "A1",
+        end_space: "B3"
+      }
+
+      # expect(post "/api/v1/games/#{game.id}/ships", params: ship_2_payload, headers: headers).to raise_error(InvalidShipPlacement)
```
Comment (lilwillifo) [inline spec/requests/ship_placer_spec.rb]: @jmejia @icorson3 I'm having trouble edge case testing for the raising an error for invalid ship placement. I think I'm following the docs, but it won't even run this line in the test file, because it raises the error in the test itself 🙃It may have to do with making an API call, because I haven't found a good example with that.

Comment (jmejia) [reply spec/requests/ship_placer_spec.rb]: @lilwillifo Not sure if I'm understanding correctly but I think you might need to pass `expect` a block. Try something like this:

```
expect { post "/api/v1/games/#{game.id}/ships", params: ship_2_payload, headers: headers }.to raise_error(InvalidShipPlacement)
```

Some docs: https://relishapp.com/rspec/rspec-expectations/v/2-0/docs/matchers/expect-error

Let me know if that's not what you're asking.
